test/extensions/bootstrap/wasm/test_data/speed_cpp.cc

```diff
@@ -69,6 +97,42 @@ void grpc_service1000_test() {
   }
 }
 
+void modify_metadata_test() {
+  auto path = getRequestHeader(":path");
+  addRequestHeader("newheader", "newheadervalue");
+  auto server = getRequestHeader("server");
+  replaceRequestHeader("server", "envoy-wasm");
+  replaceRequestHeader("envoy-wasm", "server");
+  removeRequestHeader("newheader");
+}
+
+void modify_metadata1000_test() {
+  for (int x = 0; x < 1000; x++) {
+    auto path = getRequestHeader(":path");
+    addRequestHeader("newheader", "newheadervalue");
+    auto server = getRequestHeader("server");
```

Comment:
Are we sure that this code is not optimized away by the compiler ?
Do we use flags to prevent dead code elimination? @jplevyak

Reply:
@kyessenov ?

Reply:
yeah, it should be. i'm surprised it even compiles since it's an unused value.

Reply:
Adding in the output from this run to the pr description. Also currently blocked on this pr until 574 is merged in, since that will enable me to move some of wasm speed tests under the http filter tests so that I can make the comprehensive test @bianpengyuan mentioned above.

Reply:
I don't think it can optimize this away because there are too many side effects.  For example, getRequestheader is going to result in a malloc which is visible to the system because C++ doesn't have a semantic model which allows memory allocation or other libc calls to be elided.  The same is true of the other calls.  The compiler has no way of knowing that removeRequestHeader is idempotent or that replaceRequestHeader is as well.